src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -1270,19 +1402,21 @@ export async function getMixedRouteCandidatePools({
    *
    * This way we can reduce calls to our provider since it's possible to generate a lot of mixed routes
    */
-  /// We only really care about pools involving the tokenIn or tokenOut explictly,
-  /// since there's no way a long tail token in V2 would be routed through as an intermediary
+    /// We only really care about pools involving the tokenIn or tokenOut explictly,
+    /// since there's no way a long tail token in V2 would be routed through as an intermediary
   const V2topByTVLPoolIds = new Set(
-    [
-      ...V2candidatePools.selections.topByTVLUsingTokenIn,
-      ...V2candidatePools.selections.topByBaseWithTokenIn,
-      /// tokenOut:
-      ...V2candidatePools.selections.topByTVLUsingTokenOut,
-      ...V2candidatePools.selections.topByBaseWithTokenOut,
-      /// Direct swap:
-      ...V2candidatePools.selections.topByDirectSwapPool,
-    ].map((poolId) => poolId.id)
-  );
+      [
+        ...V2candidatePools.selections.topByTVLUsingTokenIn,
+        ...V2candidatePools.selections.topByBaseWithTokenIn,
+        /// tokenOut:
+        ...V2candidatePools.selections.topByTVLUsingTokenOut,
+        ...V2candidatePools.selections.topByBaseWithTokenOut,
+        /// Direct swap:
+        ...V2candidatePools.selections.topByDirectSwapPool,
+        // Cross Liquidity:
+        ...crossLiquidityPools.v2Pools,
```

Comment:
We need to include the v2Pools to this Set, otherwise it would be ignored.

Reply:
why do you need to add it here and at linee 1390 i am confused?